src/sentry/integrations/issue_alert_image_builder.py

```diff
@@ -35,30 +36,49 @@ def __init__(self, group: Group, provider: ExternalProviderEnum) -> None:
 
     def get_image_url(self) -> str | None:
         try:
-            image_url = None
             metrics.incr("chartcuterie.issue_alert.attempt", tags=self.tags)
-            if self.group.issue_type == PerformanceP95EndpointRegressionGroupType:
-                image_url = self._get_endpoint_regression_image_url()
-            elif self.group.issue_type == ProfileFunctionRegressionType:
-                image_url = self._get_function_regression_image_url()
-
+            image_url = cache.get(self.cache_key)
             if image_url:
                 metrics.incr("chartcuterie.issue_alert.success", tags=self.tags)
                 return image_url
+
+            lock_key = f"lock_{self.cache_key}"
+            lock_acquired = cache.add(key=lock_key, value="locked", timeout=120)
```

Comment:
@evanpurkhiser what are your thoughts about timeout and sleep here?
1. Timeout here means the lock will expire after 120s. I think it makes sense for 1 query and 1 service call but we can honestly increase this with low risk. Let's say 240 s
2. At line 61 I need to add a sleep. If lock is not acquired it means someone is creating the image right now so we wait for the image to get to cache. I don't want to put a `while True:` for obvious reason so was thinking to use the same time we use for timeout.

Reply:
I would recommend using our locking utilities https://github.com/getsentry/sentry/blob/master/src/sentry/utils/locking/lock.py

There is a blocking lock you can use that will do this for you.

Reply:
I didn't know if existed, will definitely reuse and set a timeout of 240s.

Reply:
240s feels pretty high no? How long should it take to generate these images?

Reply:
Ideally less than 1 mins but I want to set it to worse than the worst case. Here's the trade-off I'm thinking:
Side 1: If I set it at 120s and then it takes one image to get generated 125 the second thread will acquire the lock and start re-generating the image because it's not in the cache yet.
Side 2: If I set it too high, to be sure it's definitely more than any image generation, the 2nd thread will never regenerate. The only downside I see is that lock will be released later if no one releases it.

Reply:
It would probably make sense to have a timeout configured on the call to chartcuterie and our lock just matches that timeout

Reply:
It's actually reasonably low now that [I look](https://sentry.sentry.io/performance/traces/?project=1&query=span.op%3Acharts.chartcuterie.generate_chart&statsPeriod=14d). both upload and getting response spans are less than 1s.

Reply:
Yeah I would never expect this to be slow